pkg/ds/daemon_set_test.go

```diff
@@ -0,0 +1,332 @@
+package ds
+
+import (
+	"fmt"
+	"path"
+	"testing"
+	"time"
+
+	"github.com/square/p2/pkg/util"
+
+	"github.com/square/p2/pkg/kp"
+	"github.com/square/p2/pkg/kp/dsstore"
+	"github.com/square/p2/pkg/kp/dsstore/dsstoretest"
+	"github.com/square/p2/pkg/labels"
+	"github.com/square/p2/pkg/logging"
+	"github.com/square/p2/pkg/pods"
+	"github.com/square/p2/pkg/scheduler"
+	"github.com/square/p2/pkg/types"
+
+	. "github.com/anthonybishopric/gotcha"
+	ds_fields "github.com/square/p2/pkg/ds/fields"
+	klabels "k8s.io/kubernetes/pkg/labels"
+)
+
+type fakeKpStore struct {
+	manifests map[string]pods.Manifest
```

Comment:
nit: it might be neater to use a `struct` containing prefix, nodename, manifest.id as the map key so you don't have to do `path.Join`